examples/multiple_tracks.rs

```diff
@@ -0,0 +1,76 @@
+use audio_engine::{AudioEngine, WavDecoder, OggDecoder};
+use std::{
+    io::Cursor,
+    time::Instant
+};
+
+fn log_panic() {
+    let default_hook = std::panic::take_hook();
+    std::panic::set_hook(Box::new(move |info| {
+        let current = std::thread::current();
+        let thread = current.name().unwrap_or("unnamed");
+
+        let msg = if let Some(s) = info.payload().downcast_ref::<&'static str>() {
+            *s
+        } else if let Some(s) = info.payload().downcast_ref::<String>() {
+            &**s
+        } else {
+            "Box<Any>"
+        };
+
+        match info.location() {
+            Some(location) => {
+                log::error!(
+                    "thread '{}' panicked at '{}': {}:{}",
+                    thread,
+                    msg,
+                    location.file(),
+                    location.line(),
+                );
+            }
+            None => {
+                log::error!("thread '{}' panicked at '{}'", thread, msg,)
+            }
+        }
+
+        default_hook(info);
+    }));
+}
+
+#[cfg_attr(
+    target_os = "android",
+    ndk_glue::main(
+        backtrace = "on",
+        ndk_glue = "ndk_glue",
+        logger(level = "trace", tag = "audio-engine", filter = "debug")
+    )
+)]
+fn main() {
+    #[cfg(not(target_os = "android"))]
+    env_logger::init();
+
+    log_panic();
+
+    let engine = AudioEngine::new().unwrap();
+
+    let mut track1 = engine
+        .new_sound(OggDecoder::new(Cursor::new(&include_bytes!("pipe.ogg")[..])).unwrap())
+        .unwrap();
+    track1.set_loop(true);
+    track1.play();
+
+    let mut track2 = engine
+        .new_sound(WavDecoder::new(Cursor::new(&include_bytes!("sin_500hz.wav")[..])).unwrap())
```

Comment:
Is it necessary to include a new file for this example? You could not use the other sound file for this? You can also implement a Sine SoundSource (I need to remember to put this in a example):

```rust
struct Sine {
    i: u32,
}
impl audio_engine::SoundSource for Sine {
    fn sample_rate(&self) -> u32 {
        44100
    }
    fn channels(&self) -> u16 {
        1
    }
    fn reset(&mut self) {
        self.i = 0
    }
    fn write_samples(&mut self, out: &mut [i16]) -> usize {
        for o in out.iter_mut() {
            let t = self.i as f32 / self.sample_rate() as f32;
            let amplitude = (i16::max_value() / 4) as f32;
            *o = ((500.0 * std::f32::consts::TAU * t).sin() * amplitude) as i16;
            self.i += 1;
        }
        out.len()
    }
}
```

Reply:
a pure sin wave clearly demonstrates the changes to the volume so i added another file :p using a soundsource seems like a better idea because it also demonstrates how to use the `SoundSource` struct to the user, i'll make the changes right away